### DISCUSSION TITLE
How to set a featured image?

### DISCUSSION THREAD

**Discussion by RJTK** [Ice-Hazymoon/hugo-theme-luna#50] [Q&A] [answered]
I'm am a complete noob when it comes to anything web-development, but I'm putting together a blog using the Luna theme (which looks awesome, btw).

I'm having trouble getting the `featured_image` field to work properly.  In the exampleSite, `featured_image` is just given as `cover.jpg`, which is co-located with the `index.md`.  I did this in my own `.md` file, which generally displays correctly, but when I add `featured_image: cover.jpg`, where `cover.jpg` is in the same directory as is the `.md` file, Hugo throws a `nilpointer` error.

The error is `blog/themes/hugo-theme-luna/layouts/partials/article-list/components/with-cover.html:24:60": nil pointer evaluating resource.Resource.Permalink ` and occus on line 24 of that `with-cover.html` file:

```
    {{- $processImage = partial "helper/image-process" (dict "Image" $featuredImage "Site" .Site) -}}

    {{- $featuredImageUrl = print $public_cdn $featuredImage.Permalink -}}
{{- end }}
```

I believe `$featuredImage.Permalink` is `nil`.  Is it obvious what I've done wrong here?

**Comment by RJTK** [comment]
Okay, I finally managed to solve this problem just by re-naming my `.md` file to `index.md` (instead of a more descriptive name).  Why does it need to have this name?  Is that a Hugo thing?  Or a Luna theme thing?

**Comment by Ice-Hazymoon** [comment, marked as answer]
As an example: posts/my-first-post.md can be a post and does not need to be configured separately. However, it is not a Page Bundle, so it is not possible to insert local images.

You need to change it to: posts/my-first-post/index.md and then put the image in the same directory as the index.md file

Reference: https://gohugo.io/content-management/page-bundles/

This is a feature of hugo

**Reply by RJTK** [reply]
Okay I get it now, thanks a lot.
